inc/notificationtargetcommonitilobject.class.php

```diff
@@ -1320,7 +1320,7 @@ function getDataForObject(CommonDBTM $item, array $options, $simple = false) {
 
             // Check if the author need to be anonymized
             if (ITILFollowup::getById($followup['id'])->isFromSupportAgent()
-               && $is_self_service
+               && $is_self_service && $need_anonymize
```

Comment:
Maybe `$is_self_service && $need_anonymize` checks could be done first, to prevent calling `ITILFollowup::getById($timeline_data['item']['id'])` and save a DB query.

Reply:
done